### DISCUSSION TITLE
iCloud sync

### DISCUSSION THREAD

**Discussion by stefan1983** [ViennaRSS/vienna-rss#1268] [Ideas]
Hello guys,

I love vienna-rss, but since I am using two different devices (iMac and macbook pro) I would love to have an iCloud sync, so that both computers share the same last state with each other.

Would it be possible to add this feature to the app roadmap?

Thanks a lot
Stefan

**Comment by DavidMunch** [comment]
In the mean time you can just sign up for an RSS service that Vienna supports. That'll keep everything in sync.

**Comment by stefan1983** [comment]
Is there any free service available? May be to host in local network?

**Comment by DavidMunch** [comment]
I don't know of any, since Google Readers demise. 

You could host your own RSS server, but I wouldn't think it is worth it unless you like large hobby coding projects.

**Comment by me-on-gh** [comment]
Wouldn't it be possible/easier having an auto export/import of feeds so one can just use the available cloud services ?

**Comment by Eitot** [comment]
> Wouldn't it be possible/easier having an auto export/import of feeds so one can just use the available cloud services ?

I suppose you mean something like iCloud Drive, Dropbox and so forth?

The short answer is: no. Neither the RSS/Atom feed formats nor the OPML format can adequately represent the state of the Vienna feed library. Vienna currently imports just a part of the feed formats and does not export them. For exports, the OPML outline format is used. Syncing of data sets can be hard, especially if the sources change on multiple clients before a complete sync. Merge conflicts can arise and this has to be handled properly without data loss or corruption.

Vienna supports the OpenReader draft format that some feed services use. Other services with their own APIs would have to be hard-coded into Vienna.

iCloud itself is complex. It is probably easier to adopt Apple’s Core Data storage API, but that is not trivial to implement either.

**Comment by me-on-gh** [comment]
@Eitot Yes, I was thinking of just a raw file with all the subscriptions (without read state etc). I think I will be waiting for the [3.5.5 milestone](https://github.com/ViennaRSS/vienna-rss/milestone/26) and try to use the Inoreader(openreader) sync.

**Comment by barijaona** [comment]
Some free providers of synchronization services here : https://www.freshrss.org/cloud-providers.html
